spark3/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -165,6 +179,7 @@ public StagedTable stageCreate(Identifier ident, StructType schema, Transform[]
                                  Map<String, String> properties) throws TableAlreadyExistsException {
     Schema icebergSchema = SparkSchemaUtil.convert(schema);
     try {
+      // can't stage a hadoop table
```

Comment:
I think this is okay, but if staging isn't supported then a path identifier should fail instead of going ahead without using the path. Otherwise, this will attempt to create a table in the Iceberg catalog with a crazy name, which would probably fail.

Reply:
After thinking about this a little more, I think we will need to support some form of staging for Hadoop tables. Because this catalog implements the atomic operation mix-in, the staging calls will be used for all CTAS plans. Using `SupportsCatalogOptions` would mean that `save()` gets turned into a CTAS. So if we don't want the existing creates to fail, we have to support a staged table.

We can do that in a couple of ways. First, we could create a table builder based on `HadoopCatalogTableBuilder` that supports a location. Second, we could reuse the fake staged table from the session catalog (for non-Iceberg tables). I'd prefer to create a builder that can construct the transactions for path tables. We could add it to `HadoopTables`.

Reply:
> I'd prefer to create a builder that can construct the transactions for path tables. We could add it to HadoopTables.

Agreed. Added in most recent update